### PR TITLE
Wrap menu api

### PR DESCRIPTION
**Pull request by spasiu**
@jugarrit @dannytranlx @chloepouprom 

wrapping menu endpoints https://github.com/smooch/smooch/pull/1829

### CODE REVIEW

**Comment by chloepouprom** [comment]
👍

File: tests/specs/api/menu.spec.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
+import * as httpMock from '../../mocks/http';
+import { getAuthenticationHeaders } from '../../../src/utils/auth';
+import { MenuApi } from '../../../src/api/menu';
+
+
+describe('Menu API', () => {
+    const serviceUrl = 'http://some-url.com';
+    const invalidAuthErrorMessage = 'Must not use an app token for authentication.';
+    const noPropsMessage = 'Must provide props.';
+    const noItemsMessage = 'Must provide an array of items.';
```
Comment (dannytranlx) [inline tests/specs/api/menu.spec.js]: I am not sure if we should validate with message strings? Can you match them up with error/return codes?

**Comment by dannytranlx** [comment]
PR title!

👍

**Comment by dannytranlx** [comment]
Merge commit 😫

**Pull request merged**